src/tests/Search.test.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { Router } from 'react-router-dom';
+import { render, fireEvent } from '@testing-library/react';
+import { createMemoryHistory } from 'history';
+import '@testing-library/jest-dom/extend-expect';
+import App from '../App';
+import defaultSubreddit from '../helpers/constants';
+
+describe('Search', () => {
+  it('input to be defaultSubreddit when clicking on Search link', () => {
+    const history = createMemoryHistory();
+    history.push('/search/wholesomememes');
+
+    const { getByRole } = render(
+      <Router history={history}>
+        <App />
+      </Router>,
+    );
+
+    const searchInput = getByRole('textbox');
+    const searchLink = getByRole('link', { name: /Search/ });
```

Comment:
It took me a moment to understand that this is not the search button but the link in the header. Maybe you can add that info to the description of the test and rename this variable to `headerSearchLink` or so